smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/ModifiedTrait.java

```diff
@@ -148,6 +148,8 @@ private static Map<ShapeId, List<DiffStrategy>> computeDiffStrategies(Model mode
             } else if (definition.getBreakingChanges().isEmpty()) {
                 // Avoid duplicate validation events; only perform the default validation when there are no diff rules.
                 result.put(shape.getId(), DEFAULT_STRATEGIES);
+            } else if (!definition.getBreakingChanges().isEmpty()) {
```

Comment:
I think the idea is that `computeDiffStrategies` should always include a strategy for every trait in this loop. To make that more explicit in a way, don't make this an `else if`, but just an `else`.

Reply:
I think we are on the same page: it should always include a strategy if it doesn't already have breaking changes defined on the trait.